src/main/java/io/github/thebusybiscuit/slimefun4/api/items/settings/MaterialTagSetting.java

```diff
@@ -0,0 +1,76 @@
+package io.github.thebusybiscuit.slimefun4.api.items.settings;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.Material;
+import org.bukkit.Tag;
+
+import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
+
+/**
+ * This variation of {@link ItemSetting} allows you to define a default {@link Tag}.
+ * The {@link Tag} will be translated into a {@link String} {@link List} which the user
+ * can then configure as they wish.
+ * 
+ * It also validates all inputs to be a valid {@link Material}.
+ * 
+ * @author TheBusyBiscuit
+ * 
+ * @see ItemSetting
+ *
+ */
+public class MaterialTagSetting extends ItemSetting<List<String>> {
+
+    private final Tag<Material> defaultTag;
+
+    @ParametersAreNonnullByDefault
+    public MaterialTagSetting(String key, Tag<Material> defaultTag) {
+        super(key, getAsStringList(defaultTag));
+
+        this.defaultTag = defaultTag;
+    }
+
+    /**
+     * This {@link Tag} holds the default values for this {@link MaterialTagSetting}.
+     * 
+     * @return The default {@link Tag}
+     */
+    @Nonnull
+    public Tag<Material> getDefaultTag() {
+        return defaultTag;
+    }
+
+    @Override
+    public boolean validateInput(List<String> input) {
+        if (input != null) {
+            for (String value : input) {
+                Material material = Material.matchMaterial(value);
+
+                // This value is not a valid material, the setting is not valid.
```

Comment:
```suggestion
                // This value is not a valid material, the setting is invalid.
```

Reply:
That's pretty much the same thing, and since this is a "validate" method, not valid is more appropriate of a term here.